pkg/operator/controllers/previewfeature/nsgflowlogs/nsgflowlogs_test.go

```diff
@@ -0,0 +1,250 @@
+	subnetNameWorker        = "worker"
+	subnetNameWorker2       = "worker2"
+	subnetNameMaster        = "master"
+	resourceIdMaster        = "/subscriptions/" + subscriptionId + "/resourceGroups/" + vnetResourceGroup + "/providers/Microsoft.Network/virtualNetworks/" + vnetName + "/subnets/" + subnetNameMaster
+	resourceIdWorker        = "/subscriptions/" + subscriptionId + "/resourceGroups/" + vnetResourceGroup + "/providers/Microsoft.Network/virtualNetworks/" + vnetName + "/subnets/" + subnetNameWorker
+	resourceIdWorker2       = "/subscriptions/" + subscriptionId + "/resourceGroups/" + vnetResourceGroup + "/providers/Microsoft.Network/virtualNetworks/" + vnetName + "/subnets/" + subnetNameWorker2
+)
+
+func getValidPreviewFeatureInstance() *aropreviewv1alpha1.PreviewFeature {
+	return &aropreviewv1alpha1.PreviewFeature{
+		Spec: aropreviewv1alpha1.PreviewFeatureSpec{
+			NSGFlowLogs: &aropreviewv1alpha1.NSGFlowLogs{
+				Enabled:                                 false,
+				Version:                                 0,
+				NetworkWatcherID:                        "",
+				StorageAccountResourceID:                "",
+				RetentionDays:                           0,
+				TrafficAnalyticsLogAnalyticsWorkspaceID: "",
+				TrafficAnalyticsInterval:                metav1.Duration{},
+			},
+		},
+	}
+}
+
+func getValidFlowLogFeature() *mgmtnetwork.FlowLog {
+	return &mgmtnetwork.FlowLog{
+		FlowLogPropertiesFormat: &mgmtnetwork.FlowLogPropertiesFormat{
+			TargetResourceID: to.StringPtr(""),
+			Enabled:          to.BoolPtr(true),
+			StorageID:        to.StringPtr(""),
+			RetentionPolicy: &mgmtnetwork.RetentionPolicyParameters{
+				Days: to.Int32Ptr(0),
+			},
+			Format: &mgmtnetwork.FlowLogFormatParameters{
+				Type:    mgmtnetwork.JSON,
+				Version: to.Int32Ptr(0),
+			},
+			FlowAnalyticsConfiguration: &mgmtnetwork.TrafficAnalyticsProperties{
+				NetworkWatcherFlowAnalyticsConfiguration: &mgmtnetwork.TrafficAnalyticsConfigurationProperties{
+					TrafficAnalyticsInterval: to.Int32Ptr(0),
+					WorkspaceID:              to.StringPtr(""),
+				},
+			},
+			ProvisioningState: "",
+		},
+		Location: to.StringPtr(location),
+	}
+}
+
+func TestReconcileManager(t *testing.T) {
+	for _, tt := range []struct {
+		name              string
+		subnetMock        func(*mock_subnet.MockManager, *mock_subnet.MockKubeManager)
+		instance          func(*aropreviewv1alpha1.PreviewFeature)
+		flowLogClientMock func(*mock_network.MockFlowLogsClient)
+		wantErr           string
+	}{
+		{
+			name: "do not enable flow log if parameters are missing/wrong",
+			subnetMock: func(mock *mock_subnet.MockManager, kmock *mock_subnet.MockKubeManager) {
+				kmock.EXPECT().List(gomock.Any()).Return([]subnet.Subnet{
+					{
+						ResourceID: resourceIdMaster,
+					},
+					{
+						ResourceID: resourceIdWorker,
+					},
+				}, nil)
+				mock.EXPECT().Get(gomock.Any(), gomock.Any()).Return(&mgmtnetwork.Subnet{
```

Comment:
It is ok to use `gomock.Any()` for context, but for the second argument (resource id) we need an expected value. To make sure that `n.subnets.Get(...)` is being called with the right value.

Same applies to other expected mock calls.